### PR TITLE
Logging lib

### PR DESCRIPTION
**Pull request by DavidTanner**
office

@Druotic and I were discussing using a standardized logger, possibly using bunyan and debug, but ultimately the logs won't be going through any processors so debug looks the best.  This just replaces all of the `console` commands with debug logs.  `info`, `warn`, and `error` will always print out to the console, with `debug` relying on the `DEBUG` environment variable being set.  

I will do a major release with approval.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -361,5 +344,29 @@ You will also find the following intermediate files:
 
 ## Debugging
 
-To debug some docker logs, set `DEBUG_DOCKER=true`.
-To debug the localstack docker container startup, set `DEBUG_LOCALSTACK=true`.
+To enable debug level logging we are using the [debug][debug] library to create the log lines.  
+There is also an environment variable `ENABLE_LAMBDA_LOGGING` available to
```
Comment (Druotic) [inline README.md]: I think this env var can be removed now? There are some `afterEach` hooks below that reference and I think they can be cleaned up too.

File: README.md [hunk 2]
```diff
@@ -361,5 +344,29 @@ You will also find the following intermediate files:
...
+print the results of a lambda run, both `stdout` and `stderr`
+Available flags are 
+- `lambda-tools:lambda`
+- `lambda-tools:docker`
```
Comment (Druotic) [inline README.md]: I think there are `localstack` namespaced debug level logs as well.

File: README.md (hunk 1 above)
Comment (DavidTanner) [reply README.md]: Oops, I forgot to go update these.

File: README.md (hunk 2 above)
Comment (DavidTanner) [reply README.md]: Adding that.  Thanks for the catch.

File: src/localstack.js [hunk 3]
```diff
@@ -10,11 +10,12 @@ const { buildConnectionAndConfig, waitForReady } = require('./utils/awsUtils');
 const { Writable } = require('stream');
 const logger = require('./utils/logging').getLogger('localstack');
 
-class TempWriteBuffer extends Writable {
+class LocalstackWriteBuffer extends Writable {
   constructor (resolve, container) {
     super();
-    this.reset();
```
Comment (Druotic) [inline src/localstack.js]: It looks like this may have been the last usage of `reset` - perhaps it could be deleted?

File: src/localstack.js [hunk 4]
```diff
@@ -23,19 +24,18 @@ class TempWriteBuffer extends Writable {
...
     const asBuffer = Buffer.from(chunk, 'utf8');
-    const asString = asBuffer.toString('utf8');
-    if (this._buffer) {
+    const asString = asBuffer.toString('utf8').replace(/\r/, '').trim();
+    const logs = asString.split('\n');
+    this._buffer.push(...logs);
+    if (!this._isSetUp) {
       this.logger.debug(asString);
-      this._buffer.push(asBuffer);
-      const logs = this.toString('utf8').trim().split('\n');
```
Comment (Druotic) [inline src/localstack.js]: Similar to above - I think this was the only usage of the function. Can it be deleted?

Comment (DavidTanner) [reply src/localstack.js]: I was actually thinking about moving it out to be exposed in the test context, then results from Lambda could be exposed like the current docker lambda instance.

File: src/localstack.js (hunk 3 above)
Comment (DavidTanner) [reply src/localstack.js]: If the buffer is exposed, it would make sense to allow a client to reset the buffer on demand.

**Pull request merged**